### DISCUSSION TITLE
Read from file and write to Decoder

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#764] [Q&A] [answered]
i want to play from file in this way . read bytes and write to decoder . but this don't work . 

How to make it work by reading from file and writing to decoder without copystream ? 

below is pseudo code  
EncodedAudioStream decStreammp3(i2s, mp3decoder); 
..
 loop
    { 
       int tt=audioFile.read( (uint8_t *)buff,200 ); 
       decStreammp3.write((uint8_t *)buff , tt);
     }

**Comment by pschatzmann** [comment, marked as answer]
I don't see any reason why this should not work - if you send it valid data
